metagraph/src/cli/load/load_graph.cpp

```diff
@@ -71,5 +71,16 @@ std::shared_ptr<DeBruijnGraph> load_critical_dbg(const std::string &filename) {
     exit(1);
 }
 
+std::shared_ptr<CanonicalDBG> primary_to_canonical(std::shared_ptr<DeBruijnGraph> graph,
+                                                   size_t cache_size) {
+    if (graph->get_mode() != DeBruijnGraph::PRIMARY) {
+        logger->error("Only primary mode graphs can be wrapped into canonical mode.");
+        exit(1);
+    }
+
+    logger->trace("Primary graph wrapped into canonical");
+    return std::make_shared<CanonicalDBG>(graph, cache_size);
+}
```

Comment:
Why not return `CanonicalDBG`? You can always move it to create a new shared pointer from it.

Reply:
So in such cases, it's always better to return the object itself and not a shared pointer. If you follow this rule, it will be easier to see further simplifications.